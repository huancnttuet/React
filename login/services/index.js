export { default as serverServices } from "./user";
