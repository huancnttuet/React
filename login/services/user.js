var user = require("../model/user.js");
var nodemailer = require("nodemailer");

module.exports = {
  signin: async (usernameSignIn, pwdSignIn) => {
    var valueSignIn = await user.checkSignIn(usernameSignIn, pwdSignIn);

    if (valueSignIn !== 0) {
      console.log("login true");
      return {
        code: "SUCCESS",
        message: "Đăng nhập thành công",
        result: {
          username: valueSignIn.username,
          id: valueSignIn.id
        }
      };
    } else {
      console.log("login false");
      return {
        code: "ERROR",
        message: "Sai mật khẩu"
      };
    }
  },
  signup: async (emailSignUp, usernameSignUp) => {
    const sendMail = pwd => {
      var transporter = nodemailer.createTransport({
        service: "Gmail",
        auth: {
          user: "huancnttuet@gmail.com",
          pass: "341997mok"
        }
      });

      var mailOptions = {
        from: "huancnttuet@gmail.com",
        to: "huancnttmta@gmail.com",
        subject: "Password demo",
        text: `Yourpassword: ${pwd}`
      };

      transporter.sendMail(mailOptions, function(error, info) {
        if (error) {
          console.log(error);
        } else {
          console.log("Email sent: " + info.response);
        }
      });
    };
    const createPwdSignUp = () => {
      var length = 8,
        charset =
          "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789",
        retVal = "";
      for (var i = 0, n = charset.length; i < length; ++i) {
        retVal += charset.charAt(Math.floor(Math.random() * n));
      }
      return retVal;
    };

    var checkSU = await user.checkSignUp(emailSignUp, usernameSignUp);
    console.log(checkSU);
    if (!checkSU) {
      var pwdSignUp = createPwdSignUp();

      if (await user.createUser(emailSignUp, usernameSignUp, pwdSignUp)) {
        sendMail(pwdSignUp);
        return {
          code: "SUCCESS",
          message: "create new user success"
        };
      } else {
        return {
          code: "ERROR",
          message: "error create new user"
        };
      }
    } else {
      return {
        code: "EXIST",
        message: "user or mail exist"
      };
    }
  },
  changepwd: async (id, pwd, pwdNew) => {
    if (await user.checkIdPwd(id, pwd)) {
      if (await user.changePwd(id, pwdNew)) {
        return {
          code: "SUCCESS",
          message: "Đổi mật khẩu thành công"
        };
      } else {
        return {
          code: "ERROR",
          message: "Đã có lỗi xảy ra"
        };
      }
    } else {
      return {
        code: "SAIPASS",
        message: "Sai mật khẩu"
      };
    }
  },
  forgottenacc: async emailFA => {
    var pwd = await user.checkEmailFA(emailFA);
    const sendMail = pwd => {
      var transporter = nodemailer.createTransport({
        service: "Gmail",
        auth: {
          user: "huancnttuet@gmail.com",
          pass: "341997mok"
        }
      });

      var mailOptions = {
        from: "huancnttuet@gmail.com",
        to: "huancnttmta@gmail.com",
        subject: "Forgotten Password demo",
        text: `Yourpassword: ${pwd}`
      };

      transporter.sendMail(mailOptions, function(error, info) {
        if (error) {
          console.log(error);
          return {
            code: "ERROR",
            message: "Lỗi trong quá trình gửi mail"
          };
        } else {
          console.log("Email sent: Quen pass " + info.response);
          return {
            code: "SUCCESS",
            message: "Pass đã được gửi lại về mail của bạn"
          };
        }
      });
    };
    if (pwd !== null) {
      sendMail(pwd);
    } else {
      return {
        code: "NOTFOUND",
        message: "Mail chưa được đăng ký"
      };
    }
  }
};
