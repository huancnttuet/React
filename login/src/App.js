import React from 'react';
import './css/App.css';

import TopPage from './TopPage'
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import Home from './Home'
import SignIn from './SignIn'
import SignUp from './SignUp'

function App() {
  return (
    <Router>
       <div>
         <TopPage />

         <div >
           <Route exact path="/" component={Home} />
           <Route path="/signin/:id" component={SignIn} />
           <Route path="/signup" component={SignUp} />
         </div>
       </div>


     </Router>
  );
}

export default App;
