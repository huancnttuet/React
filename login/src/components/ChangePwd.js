import React, {useState} from 'react';
import TopPage from './TopPage'
import {Form, Button, Col, Row, Container} from 'react-bootstrap'
import axios from 'axios'
import {connect} from 'react-redux'


function ChangePwd(props) {

  const pwd = useFormInput('')
  const newPwd = useFormInput('')
  const reNewPwd = useFormInput('')
  const [message, setMessage] = useState('')

  console.log(props);
  function handleClick() {
    if(pwd.value === ''){
      setMessage('Chưa nhập password')
    } else{
      if(newPwd.value === ''){
        setMessage('Chưa nhập New password')
      } else {
        if(newPwd.value !== reNewPwd.value){
          setMessage('password không khớp nhau')
        } else {
          var data = {
            id: props.id,
            pwd: pwd.value,
            pwdNew: newPwd.value
          }
          console.log(data);
          axios.post('http://localhost:8000/changepwd', {data}).then((res) => {
            console.log(res.data);
            setMessage(res.data.message)
          })
        }
      }
    }
  }

  return (
    <div>
      <TopPage type='logout' />
      <Container>
        <Row style={{marginTop:50}}>
          <Col></Col>
          <Col>
            <Form>
              <Form.Group controlId="formBasicChangePwd" {...pwd}>
                <Form.Label>Password</Form.Label>
                <Form.Control type="password" placeholder="Enter password" />
              </Form.Group>

              <Form.Group controlId="formBasicPassword1" {...newPwd}>
                <Form.Label>New Password</Form.Label>
                <Form.Control type="password" placeholder="Enter new password"/>
              </Form.Group>

              <Form.Group controlId="formBasicPassword2" {...reNewPwd}>
                <Form.Label>Re-New Password</Form.Label>
                <Form.Control type="password" placeholder="Enter re-password"/>
              </Form.Group>

              <Button variant="primary"  onClick={handleClick}>
                Submit
              </Button>
            </Form>
            <p>{message}</p>
          </Col>
          <Col></Col>
        </Row>
      </Container>
    </div>
  )
}

function useFormInput(initial){
  const [value, setValue] = useState(initial)

  function handleChange(e){
    setValue(e.target.value)
  }

  return {
    value,
    onChange: handleChange
  }
}

const mapStateToProps = state => ({
  id: state.id
})

export default connect(mapStateToProps)(ChangePwd)
