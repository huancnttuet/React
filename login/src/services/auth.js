import axios from 'axios'

const login = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/signin`, params);
}

const signup = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/signup`, params);
}

const changepwd = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/changepwd`, params);
}

const forgottenacc = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/forgottenacc`, params);
}

export default {
  login,
  signup,
  changepwd,
  forgottenacc,
}
