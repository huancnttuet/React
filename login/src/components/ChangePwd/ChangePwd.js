import React, {useState, useGlobal} from 'reactn';
import {Form, Button, Col, Row, Container} from 'react-bootstrap'
import { authServices } from 'services'

function ChangePwd(props) {
  const [id, setId] = useGlobal('id')
  const pwd = useFormInput('')
  const newPwd = useFormInput('')
  const reNewPwd = useFormInput('')
  const [message, setMessage] = useState('')

  console.log(props);
  function handleClick() {
    if(pwd.value === ''){
      setMessage('Chưa nhập password')
    } else{
      if(newPwd.value === ''){
        setMessage('Chưa nhập New password')
      } else {
        if(newPwd.value !== reNewPwd.value){
          setMessage('password không khớp nhau')
        } else {
          var data = {
            id: id,
            pwd: pwd.value,
            pwdNew: newPwd.value
          }
          console.log(data);
          authServices.changepwd(data).then((res) => {
            console.log(res.data);
            setMessage(res.data.message)
          })
        }
      }
    }
  }

  return (
    <div>
      <Container>
        <Row style={{marginTop:50}}>
          <Col></Col>
          <Col>
            <Form>
              <Form.Group controlId="formBasicChangePwd" {...pwd}>
                <Form.Label>Password</Form.Label>
                <Form.Control type="password" placeholder="Enter password" />
              </Form.Group>

              <Form.Group controlId="formBasicPassword1" {...newPwd}>
                <Form.Label>New Password</Form.Label>
                <Form.Control type="password" placeholder="Enter new password"/>
              </Form.Group>

              <Form.Group controlId="formBasicPassword2" {...reNewPwd}>
                <Form.Label>Re-New Password</Form.Label>
                <Form.Control type="password" placeholder="Enter re-password"/>
              </Form.Group>

              <Button variant="primary" id='cp-btn' onClick={handleClick}>
                Submit
              </Button>
            </Form>
            <p>{message}</p>
          </Col>
          <Col></Col>
        </Row>
      </Container>
    </div>
  )
}

function useFormInput(initial){
  const [value, setValue] = useState(initial)

  function handleChange(e){
    setValue(e.target.value)
  }

  return {
    value,
    onChange: handleChange
  }
}


export default ChangePwd
