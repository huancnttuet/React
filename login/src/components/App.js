import React, {useReducer,  setGlobal} from 'reactn';
import './css/App.css';

import { BrowserRouter as Router, Route } from "react-router-dom";
import Home from './Home'
import SignIn from './SignIn'
import SignUp from './SignUp'
import ChangePwd from './ChangePwd'
import ForgottenAcc from './ForgottenAcc'
import NotFound from './NotFound'
import TopPage from './TopPage'

const reducer = (state, action) => {
  switch (action.type) {
    case 'LOGIN':
      localStorage.setItem('authenticate', true)
      localStorage.setItem('id', action.payload)
      return {authenticate: true, id: action.payload}
    case 'LOGOUT':
      localStorage.setItem('authenticate', false)
      localStorage.setItem('id', 0)
      return {authenticate: false, id: 0}
    default:
      return state
  }
}


function App(props) {

  const [state, dispatch] = useReducer(reducer, {authenticate:localStorage.getItem('authenticate'), id: localStorage.getItem('id')})
  setGlobal({
    state: state,
    dispatch: dispatch
  })

  var pathHome
  if(localStorage.getItem('id') === 0){
    pathHome = '/'
  } else {
    pathHome = `/${localStorage.getItem('id')}`
  }

  console.log(pathHome);
  return (
    <Router>
        <TopPage />
         <div >
          <Route exact path={pathHome} component={Home} />
          <Route exact path='/' component={() => <SignIn authenticate={localStorage.getItem('authenticate')} /> } />
          <Route path="/signup" component={SignUp} />
          <Route path='/changepwd' component={ChangePwd} />
          <Route path="/forgottenacc" component={ForgottenAcc} />
         </div>

     </Router>
  );
}


export default App;
