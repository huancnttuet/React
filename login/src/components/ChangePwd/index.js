import ChangePwd from './ChangePwd'

export default ChangePwd
