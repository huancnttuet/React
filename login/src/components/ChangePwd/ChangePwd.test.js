import React from 'react';
import { shallow, mount } from 'enzyme';
import App from '../App';
import ChangePwd from './ChangePwd';
import { authServices } from 'services'
import axios from 'axios'

let wrapper
beforeEach(() => {
    wrapper = mount(<App />);
});

afterEach(() => {
  wrapper.unmount();
})

describe('ChangePwd', () => {
  it('click button ', () => {
    const loginMock = jest.spyOn(authServices, 'changepwd')
    const wrapper1 = mount(<ChangePwd />);
    expect(wrapper1.debug()).toMatchSnapshot();
    wrapper1.find('#formBasicChangePwd').simulate('change', {target: {
      value: '12'
    }})
    wrapper1.find('#formBasicPassword1').simulate('change', {target: {
      value: '1'
    }})
    wrapper1.find('#formBasicPassword2').simulate('change', {target: {
      value: '1'
    }})
    wrapper1.find('#cp-btn').at(0).simulate('click')
    wrapper1.find('#cp-btn').at(1).simulate('click')
    expect(loginMock).toHaveBeenCalled();
  });
  test('test API', () => {
    const data = {
      id: '4',
      pwd: '12',
      pwdNew: '1',
    }
    return authServices.changepwd(data).then(data => {
      expect(data.data).toMatchObject({code: 'SAIPASS'});
    });
  });

});
