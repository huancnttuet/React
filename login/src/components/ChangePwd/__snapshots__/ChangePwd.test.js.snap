// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePwd click button  1`] = `
"<ChangePwd>
  <div>
    <Container as=\\"div\\" fluid={false}>
      <div className=\\"container\\">
        <Bootstrap(Row) style={{...}}>
          <Row style={{...}} bsPrefix=\\"row\\" as=\\"div\\" noGutters={false}>
            <div style={{...}} className=\\"row\\">
              <Col as=\\"div\\">
                <div className=\\"col\\" />
              </Col>
              <Col as=\\"div\\">
                <div className=\\"col\\">
                  <Form inline={false} as=\\"form\\">
                    <form className=\\"\\">
                      <FormGroup controlId=\\"formBasicChangePwd\\" value=\\"\\" onChange={[Function: handleChange]} as=\\"div\\">
                        <div value=\\"\\" onChange={[Function: handleChange]} className=\\"form-group\\">
                          <FormLabel column={false} srOnly={false}>
                            <label htmlFor=\\"formBasicChangePwd\\" className=\\"form-label\\">
                              Password
                            </label>
                          </FormLabel>
                          <FormControl type=\\"password\\" placeholder=\\"Enter password\\" as=\\"input\\">
                            <input placeholder=\\"Enter password\\" type=\\"password\\" readOnly={[undefined]} id=\\"formBasicChangePwd\\" className=\\"form-control\\" />
                          </FormControl>
                        </div>
                      </FormGroup>
                      <FormGroup controlId=\\"formBasicPassword1\\" value=\\"\\" onChange={[Function: handleChange]} as=\\"div\\">
                        <div value=\\"\\" onChange={[Function: handleChange]} className=\\"form-group\\">
                          <FormLabel column={false} srOnly={false}>
                            <label htmlFor=\\"formBasicPassword1\\" className=\\"form-label\\">
                              New Password
                            </label>
                          </FormLabel>
                          <FormControl type=\\"password\\" placeholder=\\"Enter new password\\" as=\\"input\\">
                            <input placeholder=\\"Enter new password\\" type=\\"password\\" readOnly={[undefined]} id=\\"formBasicPassword1\\" className=\\"form-control\\" />
                          </FormControl>
                        </div>
                      </FormGroup>
                      <FormGroup controlId=\\"formBasicPassword2\\" value=\\"\\" onChange={[Function: handleChange]} as=\\"div\\">
                        <div value=\\"\\" onChange={[Function: handleChange]} className=\\"form-group\\">
                          <FormLabel column={false} srOnly={false}>
                            <label htmlFor=\\"formBasicPassword2\\" className=\\"form-label\\">
                              Re-New Password
                            </label>
                          </FormLabel>
                          <FormControl type=\\"password\\" placeholder=\\"Enter re-password\\" as=\\"input\\">
                            <input placeholder=\\"Enter re-password\\" type=\\"password\\" readOnly={[undefined]} id=\\"formBasicPassword2\\" className=\\"form-control\\" />
                          </FormControl>
                        </div>
                      </FormGroup>
                      <Button variant=\\"primary\\" id=\\"cp-btn\\" onClick={[Function: handleClick]} active={false} disabled={false} type=\\"button\\">
                        <button id=\\"cp-btn\\" onClick={[Function: handleClick]} disabled={false} type=\\"button\\" className=\\"btn btn-primary\\">
                          Submit
                        </button>
                      </Button>
                    </form>
                  </Form>
                  <p />
                </div>
              </Col>
              <Col as=\\"div\\">
                <div className=\\"col\\" />
              </Col>
            </div>
          </Row>
        </Bootstrap(Row)>
      </div>
    </Container>
  </div>
</ChangePwd>"
`;
